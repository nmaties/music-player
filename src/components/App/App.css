.table-black {
    background-color: black;
}

table.table-black {
    margin-bottom: 0;
    border-width: thin;
    border-color: orange;
}

table.table-black > thead > tr > th {
    border-width: thin;
    border-color: orange;

}

table.table-black > tbody > tr > td {
    border-width: thin;
    border-color: orange;
}

.orange-input {
    background-color: orange !important;
    border-color: darkorange !important;
}

.text-darkorange {
    color: darkorange;
}