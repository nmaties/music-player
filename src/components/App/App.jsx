import React, { Component } from 'react';
import './App.css';
import songsList from '../../utilities/songsList';
import AudioPlayer from '../AudioPlayer/AudioPlayer';

class App extends Component {
  constructor(props) {
    super(props);

    this.state = {
      songsItems: songsList
    };
  }

  filterBy = (evt) =>{
    let value = evt.target.value;
    let filterBy = evt.target.name;
    let filteredList = songsList.filter((el) => {
      return el[filterBy].toLowerCase().indexOf(value) !== -1;
    });

    this.setState({
      songsItems: filteredList
    });
  };

  render() {
    return (
      <div className="container">
        <br/>
        <div className="card border-orange">
          <div className="card-header bg-warning text-dark">
            <em>Songs Playlist</em>
          </div>
            <table className="table table-hover table-bordered table-black table-border-radius table-fixed text-warning">
              <thead className="text-darkorange">
              <tr className="text-center">
                <th>#</th>
                <th>
                  <input type="text"
                           name="artist"
                           placeholder="Artist"
                           className="form-control orange-input"
                           onChange={this.filterBy}/></th>
                <th>Song <i className="fas fa-music"></i></th>
                <th>
                  <input type="text"
                         name="album"
                         placeholder="Album"
                         className="form-control orange-input"
                         onChange={this.filterBy}/>
                </th>
                <th>Release Year</th>
                <th>Format</th>
                <th className="text-center">Play/Pause</th>
              </tr>
              </thead>
              <tbody className="pagination-container">
              {this.state.songsItems.map((el, idx) => {
                el.songLink = 'http://docs.google.com/uc?export=open&id=' + el.songLink;
                return (
                  <tr className="text-center" key={idx}>
                    <td>{idx + 1}</td>
                    <td>{el.artist}</td>
                    <td>{el.songName}</td>
                    <td>{el.album}</td>
                    <td>{el.releaseYear}</td>
                    <td>{el.format}</td>
                    <td><AudioPlayer song={el} key={el.id}/></td>
                  </tr>
                )
              })}
              </tbody>
            </table>
        </div>
      </div>
    );
  }
}

export default App;
