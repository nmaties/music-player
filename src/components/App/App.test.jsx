import App from './App.jsx';
import React from 'react';
import renderer from 'react-test-renderer';

describe('App', () => {
  const songsList = [
    {
      artist: 'test 1',
      songName: 'test 1',
      format: '.mp3',
      releaseYear: 2018,
      album: 'test 1',
      songLink: 'sadas231231sa',
      id: 1
    },
    {
      artist: 'test 2',
      songName: 'test 2',
      format: '.mp3',
      releaseYear: 2018,
      album: 'test 2',
      songLink: 'sxaxasda31231',
      id: 1
    },
    {
      artist: 'test 3',
      songName: 'test 3',
      format: '.mp3',
      releaseYear: 2018,
      album: 'test 3',
      songLink: 'qeqweqsqdq',
      id: 1
    }
  ];
  it('is displayed fine', () => {
    const app = renderer
      .create(<App/>);

    app.getInstance().setState({
      songsItems: songsList
    });
    expect(app.toJSON()).toMatchSnapshot();
  });


  it('is displayed fine when filtering by artist', () => {
    const app = renderer
      .create(<App/>);

    app.getInstance().setState({
      songsItems: songsList.filter((el)=> el.artist === 'test 1')
    });
    expect(app.toJSON()).toMatchSnapshot();
  });

  it('is displayed fine when filtering by album', () => {
    const app = renderer
      .create(<App/>);

    app.getInstance().setState({
      songsItems: songsList.filter((el)=> el.album === 'test 2')
    });
    expect(app.toJSON()).toMatchSnapshot();
  });

});

