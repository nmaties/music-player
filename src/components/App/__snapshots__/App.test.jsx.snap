// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App is displayed fine 1`] = `
<div
  className="container"
>
  <br />
  <div
    className="card border-orange"
  >
    <div
      className="card-header bg-warning text-dark"
    >
      <em>
        Songs Playlist
      </em>
    </div>
    <table
      className="table table-hover table-bordered table-black table-border-radius table-fixed text-warning"
    >
      <thead
        className="text-darkorange"
      >
        <tr
          className="text-center"
        >
          <th>
            #
          </th>
          <th>
            <input
              className="form-control orange-input"
              name="artist"
              onChange={[Function]}
              placeholder="Artist"
              type="text"
            />
          </th>
          <th>
            Song 
            <i
              className="fas fa-music"
            />
          </th>
          <th>
            <input
              className="form-control orange-input"
              name="album"
              onChange={[Function]}
              placeholder="Album"
              type="text"
            />
          </th>
          <th>
            Release Year
          </th>
          <th>
            Format
          </th>
          <th
            className="text-center"
          >
            Play/Pause
          </th>
        </tr>
      </thead>
      <tbody
        className="pagination-container"
      >
        <tr
          className="text-center"
        >
          <td>
            1
          </td>
          <td>
            test 1
          </td>
          <td>
            test 1
          </td>
          <td>
            test 1
          </td>
          <td>
            2018
          </td>
          <td>
            .mp3
          </td>
          <td>
            <div>
              <audio>
                <source
                  src="http://docs.google.com/uc?export=open&id=sadas231231sa"
                />
              </audio>
              <div
                className="col-md-12 row"
              >
                <div
                  className="audio-controls col-md-6"
                >
                  <span
                    className="btn btn-default"
                  >
                    <i
                      className="fas fa-play "
                      onClick={[Function]}
                    />
                  </span>
                  <span
                    className="btn btn-default"
                  >
                    <i
                      className="fas fa-pause filled-icon"
                      onClick={[Function]}
                    />
                  </span>
                </div>
                <div
                  className="col-md-6"
                >
                  <span>
                    <input
                      className="slider"
                      max="100"
                      min="0"
                      name="volume"
                      onChange={[Function]}
                      step="1"
                      type="range"
                    />
                  </span>
                </div>
              </div>
            </div>
          </td>
        </tr>
        <tr
          className="text-center"
        >
          <td>
            2
          </td>
          <td>
            test 2
          </td>
          <td>
            test 2
          </td>
          <td>
            test 2
          </td>
          <td>
            2018
          </td>
          <td>
            .mp3
          </td>
          <td>
            <div>
              <audio>
                <source
                  src="http://docs.google.com/uc?export=open&id=sxaxasda31231"
                />
              </audio>
              <div
                className="col-md-12 row"
              >
                <div
                  className="audio-controls col-md-6"
                >
                  <span
                    className="btn btn-default"
                  >
                    <i
                      className="fas fa-play "
                      onClick={[Function]}
                    />
                  </span>
                  <span
                    className="btn btn-default"
                  >
                    <i
                      className="fas fa-pause filled-icon"
                      onClick={[Function]}
                    />
                  </span>
                </div>
                <div
                  className="col-md-6"
                >
                  <span>
                    <input
                      className="slider"
                      max="100"
                      min="0"
                      name="volume"
                      onChange={[Function]}
                      step="1"
                      type="range"
                    />
                  </span>
                </div>
              </div>
            </div>
          </td>
        </tr>
        <tr
          className="text-center"
        >
          <td>
            3
          </td>
          <td>
            test 3
          </td>
          <td>
            test 3
          </td>
          <td>
            test 3
          </td>
          <td>
            2018
          </td>
          <td>
            .mp3
          </td>
          <td>
            <div>
              <audio>
                <source
                  src="http://docs.google.com/uc?export=open&id=qeqweqsqdq"
                />
              </audio>
              <div
                className="col-md-12 row"
              >
                <div
                  className="audio-controls col-md-6"
                >
                  <span
                    className="btn btn-default"
                  >
                    <i
                      className="fas fa-play "
                      onClick={[Function]}
                    />
                  </span>
                  <span
                    className="btn btn-default"
                  >
                    <i
                      className="fas fa-pause filled-icon"
                      onClick={[Function]}
                    />
                  </span>
                </div>
                <div
                  className="col-md-6"
                >
                  <span>
                    <input
                      className="slider"
                      max="100"
                      min="0"
                      name="volume"
                      onChange={[Function]}
                      step="1"
                      type="range"
                    />
                  </span>
                </div>
              </div>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`App is displayed fine when filtering by album 1`] = `
<div
  className="container"
>
  <br />
  <div
    className="card border-orange"
  >
    <div
      className="card-header bg-warning text-dark"
    >
      <em>
        Songs Playlist
      </em>
    </div>
    <table
      className="table table-hover table-bordered table-black table-border-radius table-fixed text-warning"
    >
      <thead
        className="text-darkorange"
      >
        <tr
          className="text-center"
        >
          <th>
            #
          </th>
          <th>
            <input
              className="form-control orange-input"
              name="artist"
              onChange={[Function]}
              placeholder="Artist"
              type="text"
            />
          </th>
          <th>
            Song 
            <i
              className="fas fa-music"
            />
          </th>
          <th>
            <input
              className="form-control orange-input"
              name="album"
              onChange={[Function]}
              placeholder="Album"
              type="text"
            />
          </th>
          <th>
            Release Year
          </th>
          <th>
            Format
          </th>
          <th
            className="text-center"
          >
            Play/Pause
          </th>
        </tr>
      </thead>
      <tbody
        className="pagination-container"
      >
        <tr
          className="text-center"
        >
          <td>
            1
          </td>
          <td>
            test 2
          </td>
          <td>
            test 2
          </td>
          <td>
            test 2
          </td>
          <td>
            2018
          </td>
          <td>
            .mp3
          </td>
          <td>
            <div>
              <audio>
                <source
                  src="http://docs.google.com/uc?export=open&id=http://docs.google.com/uc?export=open&id=sxaxasda31231"
                />
              </audio>
              <div
                className="col-md-12 row"
              >
                <div
                  className="audio-controls col-md-6"
                >
                  <span
                    className="btn btn-default"
                  >
                    <i
                      className="fas fa-play "
                      onClick={[Function]}
                    />
                  </span>
                  <span
                    className="btn btn-default"
                  >
                    <i
                      className="fas fa-pause filled-icon"
                      onClick={[Function]}
                    />
                  </span>
                </div>
                <div
                  className="col-md-6"
                >
                  <span>
                    <input
                      className="slider"
                      max="100"
                      min="0"
                      name="volume"
                      onChange={[Function]}
                      step="1"
                      type="range"
                    />
                  </span>
                </div>
              </div>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`App is displayed fine when filtering by artist 1`] = `
<div
  className="container"
>
  <br />
  <div
    className="card border-orange"
  >
    <div
      className="card-header bg-warning text-dark"
    >
      <em>
        Songs Playlist
      </em>
    </div>
    <table
      className="table table-hover table-bordered table-black table-border-radius table-fixed text-warning"
    >
      <thead
        className="text-darkorange"
      >
        <tr
          className="text-center"
        >
          <th>
            #
          </th>
          <th>
            <input
              className="form-control orange-input"
              name="artist"
              onChange={[Function]}
              placeholder="Artist"
              type="text"
            />
          </th>
          <th>
            Song 
            <i
              className="fas fa-music"
            />
          </th>
          <th>
            <input
              className="form-control orange-input"
              name="album"
              onChange={[Function]}
              placeholder="Album"
              type="text"
            />
          </th>
          <th>
            Release Year
          </th>
          <th>
            Format
          </th>
          <th
            className="text-center"
          >
            Play/Pause
          </th>
        </tr>
      </thead>
      <tbody
        className="pagination-container"
      >
        <tr
          className="text-center"
        >
          <td>
            1
          </td>
          <td>
            test 1
          </td>
          <td>
            test 1
          </td>
          <td>
            test 1
          </td>
          <td>
            2018
          </td>
          <td>
            .mp3
          </td>
          <td>
            <div>
              <audio>
                <source
                  src="http://docs.google.com/uc?export=open&id=http://docs.google.com/uc?export=open&id=sadas231231sa"
                />
              </audio>
              <div
                className="col-md-12 row"
              >
                <div
                  className="audio-controls col-md-6"
                >
                  <span
                    className="btn btn-default"
                  >
                    <i
                      className="fas fa-play "
                      onClick={[Function]}
                    />
                  </span>
                  <span
                    className="btn btn-default"
                  >
                    <i
                      className="fas fa-pause filled-icon"
                      onClick={[Function]}
                    />
                  </span>
                </div>
                <div
                  className="col-md-6"
                >
                  <span>
                    <input
                      className="slider"
                      max="100"
                      min="0"
                      name="volume"
                      onChange={[Function]}
                      step="1"
                      type="range"
                    />
                  </span>
                </div>
              </div>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
